sklearn/ensemble/_hist_gradient_boosting/tests/test_compare_lightgbm.py

```diff
@@ -77,14 +96,21 @@ def test_same_predictions_regression(seed, min_samples_leaf, n_samples, max_leaf
 
     pred_lightgbm = est_lightgbm.predict(X_train)
     pred_sklearn = est_sklearn.predict(X_train)
-    # less than 1% of the predictions are different up to the 3rd decimal
-    assert np.mean(abs(pred_lightgbm - pred_sklearn) > 1e-3) < 0.011
-
-    if max_leaf_nodes < 10 and n_samples >= 1000:
+    if loss in ("gamma", "poisson"):
+        # More than 65% of the predictions must be close up to the 2nd decimal.
```

Comment:
This looks a bit lax. Any idea why we can be stricter in the case of the squared error? I wouldn't require to make it stricter as a pre-requisite to merge this PR as the Poisson loss is already in `main` but lightgbm equivalence was previously just not tested.

But we could at least add a TODO comment to inform the reader that we are not 100% satisfied with the state of things :)


```suggestion
        # More than 65% of the predictions must be close up to the 2nd decimal.
        # TODO: investigate the cause of the remaining discrepancies with lightgbm.
```

Reply:
Maybe this is because, in the context of the squared error, we only test for the equivalence with the larger dataset with shallow trees (`max_leaf_nodes < 10 and n_samples >= 1000`): this condition might allow the averaging effect in leafs to mitigate the impact of rounding errors and maybe reduce the likelihood of ties when deciding on which features and thresholds to split on?

Reply:
Maybe, LightGBM's algo deviates more from ours for general losses (=not squared error). For instance in the Poisson case, LightGBM has a `poisson_max_delta_step` which I set to `1e-12`, but I'm not 100% sure of the effective difference this causes.

I'll add a comment as suggested.

Reply:
And BTW, our losses have a better test suite than the ones in LightGBM:smirk: